### ISSUE TITLE
Excerpt Block: "Add "read more" link text" link is not working

### ISSUE THREAD

**Issue by hanneslsm** [WordPress/gutenberg#52204]
### Description

It's not possible to click or do anything with the "Add "read more" link text" link in the Excerpt block.

### Step-by-step reproduction instructions

1. Create a new Excerpt block
2. Try to click the "Add "read more" link text" link

### Screenshots, screen recording, code snippet

<img width="841" alt="image" src="https://github.com/WordPress/gutenberg/assets/17861443/f6349bc9-e7e7-467e-a90c-2f2d30f2b92e">

### Environment info

- WP 6.2.2
- GB 16.1.0
- Custom theme

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by carolinan** [comment]
The placeholder is an instruction to add the link text.
Please select the link and start typing. Different themes styles links differently, a properly styled theme will have different styles depending on if the link is selected (active, focus).

This behavior of selecting and then typing is not unique to the excerpt block. It is used in for example the query pagination blocks.
The difference is that the blue line that is used to indicate the selection in the editor, is used for both the excerpt text and the link, because both these elements are part of the block.

An option is to use the excerpt block in combination with the separate "Read more" block.

**Comment by hanneslsm** [comment]
> Please select the link and start typing. 

Wha…
>  It is used in for example the query pagination blocks.

… oooooh! Wow, that was unexpected!

I think the UX can be easily improved here:
1. Change the copy to "Type to add a "Read more" link".
This is the same wording as for the paragraph block "Type / to chose a block" and therefore it's more clear the text is a placeholder. The current _"Add "read more" link text"_ sound like one should click to add.
2. Change the cursor on hover to ["text" ](https://www.w3schools.com/cssref/playit.php?filename=playcss_cursor&preval=text) (also like the paragraph block)
This should also be done for the query pagination blocks.

@annezazu Is this a UX polish ticket now? 😄

**Comment by carolinan** [comment]
> The current "Add "read more" link text" sound like one should click to add.

Yes, you need to select the link inside the block, not the block. Wether you use the keyboard, mouse click or something else.

It is not "select the block and start typing" it is literally "select the link and start typing".

**Comment by carolinan** [comment]
I still think this link should be replaced by the read more link block as an inner block but it is complicated...

**Comment by hanneslsm** [comment]
> Yes, you need to select the link inside the block, not the block.

I think this is the issue here. There is not really an visual indication that the link can even be selected (change cursor to text would be one). While recording I've seen that the "bold" option shows up, but this is too subtle. The caret is not visible because of the block border.

https://github.com/WordPress/gutenberg/assets/17861443/0eb9fa36-5607-4ff0-bc0b-62876871e655

**Comment by carolinan** [comment]
Changing to caret should not be an issue.
But all links in the editor are clickable. Some don't actually lead anywhere but are still clickable. If there is no visible status change, then that is an issue with the theme link styles.

**Comment by carolinan** [comment]
For example, here is Twenty Twenty-One. Please see the mouse pointer changing, too:

https://github.com/WordPress/gutenberg/assets/7422055/a3b67962-f9b6-4fc1-97b1-0e42847e3e04

**Comment by hanneslsm** [comment]
> For example, here is Twenty Twenty-One. Please see the mouse pointer changing, too:

Interesting! This is what I would have expected.
Looks like the **real** bug here is that the cursor doesn't change on my theme, as you can see on the recording above. In Brave and Safari.
It's a nearly blank theme, anything I should have added to the theme.js or so?

**Comment by carolinan** [comment]
I think we should look at enforcing the cursor in the editor itself. But the link style is from the theme and the theme can override both. 
There is no specific functionality for **editor** only styles in theme.json.

**Comment by hanneslsm** [comment]
> But the link style is from the theme and the theme can override both.

Not sure if I got this.
The cursor does not change, even if theme theme has a set link style.

https://github.com/WordPress/gutenberg/assets/17861443/84538750-74ae-4e27-ac5a-046658967348

**Comment by annezazu** [comment]
Just saw my @ mention and noting that it doesn't quite seem like it for UX & Polish but I love that you're keeping an eye out for opportunities to add there.

**Comment by carolinan** [comment]
> The cursor does not change, even if theme theme has a set link style.

I am not sure I understand either.
If I read your text literally, the **link** is styled. _But does the theme include CSS for the cursor itself?_  If it does, and it is not showing, then maybe the CSS has too low specificity or is targeting the wrong link...
https://developer.mozilla.org/en-US/docs/Web/CSS/cursor
https://caniuse.com/?search=cursor

**Comment by hanneslsm** [comment]
> If I read your text literally, the link is styled. But does the theme include CSS for the cursor itself? 

The theme I'm using is pretty new and the theme.js nearly empty.
No additional css for links, cursor, anything else (just some block styles).
However, the cursor does not change when hovering the _"Add "read more" link text"_ and therefore a visual feedback is missing. This made me think I could **click** the link (and something will appear or so), not that I could **select** it, so that I should start typing.

**Comment by carolinan** [comment]
I think I still don't understand... why is _clicking on_ the link and _selecting the link_ two different things..? Perhaps a recording would help.

**Comment by draganescu** [comment]
I chnaged the labels of this issue to Bug because it is an UI bug. 

https://github.com/WordPress/gutenberg/assets/107534/2d2f3d15-6789-476f-b59a-3ae0c2ba7cff

To quote another reporter:

> Normally, when you select something that allows you to type the cursor shows up where you’re going to type and the placeholder text goes away.

**Comment by shrivastavanolo** [comment]
## Reproduction Report
### Description
This report validates whether the issue can be reproduced.

### Environment
- WordPress: 6.9-alpha-60500
- PHP: 8.2.29
- Server: Apache/2.4.62 (Debian)
- Database: mysqli (Server: 11.8.2-MariaDB-ubu2404 / Client: mysqlnd 8.2.29)
- Browser: Chrome 138.0.0.0
- OS: macOS
- Theme: Twenty Twenty-Two 2.0
- MU Plugins: None activated
- Plugins:
  * Gutenberg 21.3.0-rc.1
  * Test Reports 1.2.0

### Actual Results
1.  ✅ Error condition occurs (reproduced).

### Steps for reproduction
1. Open template editor
2. Add excerpt block
3. Try to click on "Add "read more" link text" and observe

### Additional Notes
- The functionality of the "Add "read more" link text" works as expected (allows user to write a "read more" link text)
- As @draganescu mentioned it is a UI bug and will be fixed by displaying a blinking cursor where text is supposed to be typed IMO.
- Please let me know if I missed any reproduction steps!

### Supplemental Artifacts

https://github.com/user-attachments/assets/c5233a58-66e2-4de5-86ad-4671c3bd1a22
